include/deal.II/base/config.h.in

```diff
@@ -170,7 +170,7 @@ DEAL_II_NAMESPACE_CLOSE
 // Kokkos was configured with CUDA or HIP support. This is problematic
 // when calling non-constexpr functions in constexpr functions. Hence, we
 // need a way to annotate functions explicitly as host-only.
-#if (defined(__clang__) && defined(__CUDA__)) || defined(KOKKOS_ENABLE_HIP)
+#if (defined(__clang__) && defined(__CUDA__)) || @DEAL_II_KOKKOS_ENABLE_HIP@
```

Comment:
I'd rather make the configured file more readable by using
```
#cmakedefine DEAL_II_KOKKOS_ENABLE_HIP
```
and then
```suggestion
#if (defined(__clang__) && defined(__CUDA__)) || defined(DEAL_II_KOKKOS_ENABLE_HIP)
```

Reply:
Alright, I've changed it